roles/gpu_operator_deploy_from_operatorhub/tasks/deploy_clusterpolicy.yml

```diff
@@ -42,9 +42,20 @@
   retries: 20
   delay: 15
 
+- name: Optionally call the patch_clusterpolicy_role hook to modify the clusterpolicy
+  include_role:
+    name: "{{ patch_clusterpolicy_role }}"
+  when: patch_clusterpolicy_role | default('', true) | trim != ''
```

Comment:
the issue with this `include_role` design is that it makes it hard to pass parameters to the `patch` role,
eg:
```
toolbox/gpu-operator/set_repo-config.sh REPO_FILE DEST_DIR
toolbox/gpu-operator/set_repo-config.sh --rhel-beta
```
it would be hard to replicate this example without creating a new script or adding many arguments to the original one ...

I'm not sure what's the best way to organize that ....

Reply:
for this task, I think what would be the best from my POV is something like this:
```
default:
  patch_clusterpolicy_role: echo nothing to do for patching

shell: "{{ patch_clusterpolicy_role }} <  "{{ artifact_extra_logs_dir }}"/gpu_operator_clusterpolicy.json"
env: # (not ansible) 
   SRC_CLUSTER_POLICY: "{{ artifact_extra_logs_dir }}"/gpu_operator_clusterpolicy.json"
   DST_CLUSTER_POLICY: "{{ artifact_extra_logs_dir }}"/gpu_operator_clusterpolicy_patched.json"
```
this way, the patching is fully independent from the deployment, and it can receive custom parameters from the invocation command. Mind that we'll have to `unset` some ansible logging variables before calling `source _common.sh` to avoid messing up with the logging directories.